### PR TITLE
Expose the generated RuntimeClass name under the profile status

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Expose the generated RuntimeClass name under the profile status,
to give better visibility to a user what RuntimeClass should
be used under the pod spec, when latency-sensitive features
are required from the CRI-O.

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

**Comment by yuvalk** [comment]
can you include the 'why' in the commit msg body?

**Comment by fromanirh** [comment]
looks reasonnable. For lgtm, please make the commit message a little less terse like @yuvalk already suggested.

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 1]
```diff
@@ -145,7 +145,7 @@ spec:
           description: PerformanceProfileStatus defines the observed state of PerformanceProfile.
           properties:
             conditions:
-              description: conditions represents the latest available observations
+              description: Conditions represents the latest available observations
```
Comment (marcel-apf) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:148]: Not for this patch, Under the 7 rules of commits these kind of changes, even trivial, should be in a separate patch, so we can concentrate on the patch with the "real" changes.

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 2]
```diff
@@ -172,8 +172,12 @@ spec:
                 - type
                 type: object
               type: array
+            runtimeClass:
+              description: RuntimeClass contains the name of the RuntimeClass resource
+                create by the operator.
```
Comment (marcel-apf) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:177]: created?

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml [hunk 3]
```diff
@@ -172,8 +172,12 @@ spec:
                 - type
                 type: object
               type: array
+            runtimeClass:
+              description: RuntimeClass contains the name of the RuntimeClass resource
+                create by the operator.
```
Comment (marcel-apf) [inline deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml:177]: created?

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml [hunk 4]
```diff
@@ -172,8 +172,12 @@ spec:
...
+              type: string
             tuned:
-              description: tuned points to the Tuned custom resource object that contains
+              description: Tuned points to the Tuned custom resource object that contains
```
Comment (marcel-apf) [inline deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:180]: Same comment as before

File: docs/performance_profile.md [hunk 5]
```diff
@@ -126,8 +126,9 @@ PerformanceProfileStatus defines the observed state of PerformanceProfile.
 
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
-| conditions | conditions represents the latest available observations of current state. | []conditionsv1.Condition | false |
-| tuned | tuned points to the Tuned custom resource object that contains the tuning values generated by this operator. | *string | false |
+| conditions | Conditions represents the latest available observations of current state. | []conditionsv1.Condition | false |
+| tuned | Tuned points to the Tuned custom resource object that contains the tuning values generated by this operator. | *string | false |
+| runtimeClass | RuntimeClass contains the name of the RuntimeClass resource create by the operator. | *string | false |
```
Comment (marcel-apf) [inline docs/performance_profile.md:131]: also "created"

File: pkg/apis/performance/v1/performanceprofile_types.go [hunk 6]
```diff
@@ -106,12 +106,14 @@ type RealTimeKernel struct {
...
 type PerformanceProfileStatus struct {
-	// conditions represents the latest available observations of current state.
+	// Conditions represents the latest available observations of current state.
 	// +optional
 	Conditions []conditionsv1.Condition `json:"conditions,omitempty"`
-	// tuned points to the Tuned custom resource object that contains the tuning values generated by this operator.
+	// Tuned points to the Tuned custom resource object that contains the tuning values generated by this operator.
 	// +optional
 	Tuned *string `json:"tuned,omitempty"`
+	// RuntimeClass contains the name of the RuntimeClass resource create by the operator.
```
Comment (marcel-apf) [inline pkg/apis/performance/v1/performanceprofile_types.go:115]: created :)

File: pkg/controller/performanceprofile/status.go [hunk 7]
```diff
@@ -62,6 +63,12 @@ func (r *ReconcilePerformanceProfile) updateStatus(profile *performancev1.Perfor
 		modified = true
 	}
 
+	if profileCopy.Status.RuntimeClass == nil {
+		runtimeClassName := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+		profileCopy.Status.RuntimeClass = &runtimeClassName
```
Comment (marcel-apf) [inline pkg/controller/performanceprofile/status.go:68]: Can you explain more in the commit about why we set the RuntimeClass? It seems even a little explanation of "why" we set and what we do with it would be nice.

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 2 above)
Comment (fromanirh) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:177]: +1 good catch

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 1 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml:148]: done

File: deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml (hunk 2 above)
Comment (cynepco3hahue) [reply deploy/crds/performance.openshift.io_performanceprofiles_crd.yaml]: done

File: deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml (hunk 3 above)
Comment (cynepco3hahue) [reply deploy/olm-catalog/performance-addon-operator/4.6.0/performance.openshift.io_performanceprofiles_crd.yaml]: done

File: pkg/controller/performanceprofile/status.go (hunk 7 above)
Comment (cynepco3hahue) [reply pkg/controller/performanceprofile/status.go:68]: done

**Comment by cynepco3hahue** [comment]
@yuvalk @fromanirh I updated the commit message.
